lib/db2.js

```diff
@@ -575,11 +562,24 @@ DB2.prototype.getCountForAffectedRows = function(model, info) {
  */
 DB2.prototype.dropTable = function(model, cb) {
   var self = this;
-  var sql =
-      'BEGIN\nDECLARE CONTINUE HANDLER FOR SQLSTATE \'42704\'\n' +
-      'BEGIN END;\nEXECUTE IMMEDIATE \'DROP TABLE ' +
-      self.schema + '.' + self.tableEscaped(model) + '\';\nEND';
-  self.execute(sql, cb);
+  var connection;
+
+  process.nextTick(function() {
+    try {
+      connection = Driver.openSync(self.connStr);
```

Comment:
DB2 does not have the concept of DROP IF NOT EXISTS so functions that call dropTable expecting it to be successful if the table does not exist will fail.  The function must be able to execute the drop and handle the resulting error appropriately or all callers must accept that a dropTable will fail if the table does not exist.